File: solution/src/timer.cpp
#include "timer.h"

// Constructor
Timer::Timer() : startTime(), stopTime(), running(false) {}

// Reset the timer
void Timer::reset() {
  startTime = std::chrono::high_resolution_clock::time_point();
  running = false;
}

// Start the timer
void Timer::start() {
  startTime = std::chrono::high_resolution_clock::now();
  running = true;
}

// Stop the timer
void Timer::stop() {
  if (running) {
    stopTime = std::chrono::high_resolution_clock::now();
    running = false;
  }
}

// Get elapsed time in seconds
double Timer::getTime() const {
  using namespace std::chrono;
  if (running) {
    return duration<double>(high_resolution_clock::now() - startTime).count();
  } else {
    return duration<double>(stopTime - startTime).count();
  }
}

File: solution-py/solution-py/src/solver/sat/timer.py
import time

class Timer:
    def __init__(self):
        self.start_time = 0
        self.stop_time = 0
        self.running = False

    def start(self):
        self.start_time = time.time()
        self.running = True

    def stop(self):
        if self.running:
            self.stop_time = time.time()
            self.running = False

    def get_time(self):
        if self.running:
            return time.time() - self.start_time
        return self.stop_time - self.start_time
